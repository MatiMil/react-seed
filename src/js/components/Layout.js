import React from "react"
import { connect } from "react-redux"

import { fetchWelcomeText } from "../actions/welcomeTextActions"

@connect((store) => {
  return {
    welcomeText: store.welcomeText.welcomeText
  };
})
export default class Layout extends React.Component {
  componentWillMount() {
    this.props.dispatch(fetchWelcomeText())
  }

  render() {
    const { welcomeText } = this.props;

    return <div>
      <h1>{welcomeText.welcomeText}</h1>
    </div>
  }
}