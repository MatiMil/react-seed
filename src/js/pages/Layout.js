import React from "react"
import {connect} from "react-redux"
import {HashRouter, Route} from 'react-router-dom';

import {setWelcomeText, fetchText} from "../actions/welcomeTextActions"
import NavigationBar from "../components/NavigationBar"

import Index from "../pages/Index"
import Initial from "../pages/Initial"

// Require scss files
require('../../stylesheets/_all.scss');

// require LESS files
require('../../stylesheets/initial.less');

@connect((store) => {
    return {
        welcomeText: store.welcomeText.welcomeText
    };
})
export default class Layout extends React.Component {
    componentWillMount() {
        this.props.dispatch(setWelcomeText('Hello World!'));
    }

    fetchTextFromServer() {
        this.props.dispatch(fetchText());
    }

    render() {
        const {welcomeText} = this.props;
        const {location} = this.props;

        const containerStyle = {
            marginTop: "60px"
        };

        return (
            <HashRouter>
                <div>
                    <NavigationBar location={location}/>
                    <div class="container" style={containerStyle}>
                        <div class="row">
                            <div class="col-sm-12 col-md-12 col-lg-12">
                                <h1>{welcomeText.welcomeText}</h1>
                                <button onClick={this.fetchTextFromServer.bind(this)}>
                                    Fetch another text
                                </button>

                                <Route exact path="/index" component={Index}/>
                                <Route exact path="/initial" component={Initial}/>
                            </div>
                        </div>
                    </div>
                </div>
            </HashRouter>
        );
    }
}