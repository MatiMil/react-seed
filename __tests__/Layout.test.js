import React from "react"
import Layout from "../src/js/pages/Layout"
import {Provider} from "react-redux"

import store from "../src/js/store"

test('Render Layout', () => {
    const wrapper = shallow(
        <Provider store={store}>
            <Layout />
        </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
});
