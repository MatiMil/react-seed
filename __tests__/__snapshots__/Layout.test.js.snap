// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Layout 1`] = `<Connect(Layout) />`;
